docs/reference/search/rrf.asciidoc

```diff
@@ -1,6 +1,423 @@
 [[rrf]]
 === Reciprocal rank fusion
 
-Reciprocal Rank Fusion (RRF) is a simple method to combine document result sets
-from multiple queries where the queries' document scores may be unrelated.
+NOTE: Reciprocal rank fusion requires a Platinum subscription or higher.
 
+IMPORTANT: Reciprocal rank fusion is in technical preview. Features in technical
+preview may be changed or removed in a future release. Elastic will apply best
+effort to fix any issues, but features in technical preview are not subject to
+the support SLA of official GA features.
+
+Reciprocal rank fusion (RRF) is a method for combining results from
+multiple queries with different relevance indicators into a single
+result set where there is no requirement the relevance indicators for
+each query are related to each other in order to achieve high-quality results.
+
+RRF uses the following formula to determine the score for ranking each document:
+
+[source,python]
+----
+score = 0.0
+for q in queries:
+    if d in result(q):
+        score += 1.0 / ( k + rank( result(q), d ) )
+return score
+
+# where
+# k is a ranking constant
+# q is a query in the set of queries
+# d is a document in the result set of q
+# result(q) is the result set of q
+# rank( result(q), d ) is d's rank within the result(q) starting from 1
+----
+// NOTCONSOLE
+
+[[rrf-api]]
+==== Reciprocal rank fusion API
+
+RRF may be used as part of a <<search-search, search>> to combine and rank
+documents using either a single standard query and 1-to-many knn searches or no standard
+query and 2-to-many knn searches. The `rrf` parameter is an optional object defined as part
```

Comment:
```suggestion
RRF may be used as part of a <<search-search, search>> to combine and rank
documents using either a single standard query and 1-to-many kNN searches or no standard
query and 2-to-many kNN searches. The `rrf` parameter is an optional object defined as part
```

Reply:
> using either a single standard query and 1-to-many knn searches or no standard
query and 2-to-many knn searches.

Does this mean that you can only use rrf with knn queries (with optional "standard" query)? Or could it be used with multiple standard queries? If knn is required, you might want to emphasize that in the opening paragraph?

Reply:
For me this also sounds a little bit confusing. May be we need to rephrase to something like this: "You can use RRF to combine results from a query with results from a single knn search, or combine results from multiple knn searches".

Reply:
I was trying to use "standard" to differentiate from a kNN search, but I see how this is confusing. I attempted to clean this up a bit -- hopefully it makes more sense now.